dbt_semantic_interfaces/naming/dundered.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from dbt_semantic_interfaces.references import EntityReference
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+
+DUNDER = "__"
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class DunderedNameFormatter:
+    """Group by items (e.g. dimensions / entities) in a query that are named using a double underscore as a seperator.
+
+    e.g. listing__ds__week ->
+    entity_links: ["listing"]
+    element_name: "ds"
+    granularity: TimeGranularity.WEEK
```

Comment:
Can we indicate that including time granularity is a legacy anti-pattern and no implementer should be relying on it? It has caused so much trouble, and we are trying to eliminate it somehow or other.

Reply:
Added.